src/test/java/uk/gov/pay/ledger/transaction/resource/TransactionResourceIT.java

```diff
@@ -187,6 +187,148 @@ public void shouldSearchUsingAllFieldsAndReturnAllFieldsCorrectly() {
+
+        transactionFixture.insert(rule.getJdbi());
+        given().port(port)
+                .contentType(JSON)
+                .get("/v1/transaction?" +
+                        "account_id=" + gatewayAccountId +
+                        "&page=1" +
+                        "&display_size=5" +
+                        "&refund_states=success,submitted"
+                )
+                .then()
+                .statusCode(Response.Status.OK.getStatusCode())
+                .contentType(JSON)
+                .body("results[0].transaction_id", is(createdTransaction.getExternalId()))
+                .body("results[1].transaction_id", is(submittedTransaction.getExternalId()))
+                .body("results[2].transaction_id", is(nullValue()));
+    }
+
+    @Test
+    public void shouldSearchCorrectlyUsingBothPaymentAndRefundStates() {
+        String gatewayAccountId = "2";
+        TransactionFixture successfulRefund = aTransactionFixture()
+                .withTransactionType("REFUND")
+                .withState(TransactionState.SUCCESS)
+                .withDefaultCardDetails()
+                .withGatewayAccountId(gatewayAccountId)
+                .withDefaultTransactionDetails()
+                .withCreatedDate(ZonedDateTime.now())
+                .insert(rule.getJdbi());
+
+        aTransactionFixture()
+                .withTransactionType("REFUND")
+                .withState(TransactionState.SUBMITTED)
+                .withDefaultCardDetails()
+                .withGatewayAccountId(gatewayAccountId)
+                .withDefaultTransactionDetails()
+                .withCreatedDate(ZonedDateTime.now().minusHours(1))
+                .insert(rule.getJdbi());
+
+        TransactionFixture submittedPayment = aTransactionFixture()
+                .withTransactionType("PAYMENT")
+                .withState(TransactionState.SUBMITTED)
+                .withDefaultCardDetails()
+                .withGatewayAccountId(gatewayAccountId)
+                .withCreatedDate(ZonedDateTime.now().minusHours(2))
+                .withDefaultTransactionDetails()
+                .insert(rule.getJdbi());
+
+        transactionFixture.insert(rule.getJdbi());
+        given().port(port)
+                .contentType(JSON)
+                .get("/v1/transaction?" +
+                        "account_id=" + gatewayAccountId +
+                        "&page=1" +
+                        "&display_size=5" +
+                        "&refund_states=success" +
+                        "&payment_states=submitted"
+                )
+                .then()
+                .statusCode(Response.Status.OK.getStatusCode())
+                .contentType(JSON)
+                .body("results[0].transaction_id", is(successfulRefund.getExternalId()))
+                .body("results[1].transaction_id", is(submittedPayment.getExternalId()))
+                .body("results[2].transaction_id", is(nullValue()));
+    }
+
```

Comment:
In general regarding the tests it would read better if we asserted on count
``` .body("count", is(2))```
and then ids of results (but only the matching ones) rather than asserting on a null value for the third (not existing one)
```.body("results[2].transaction_id", is(nullValue()));```

In short replace ```.body("results[2].transaction_id", is(nullValue()));``` with ``` .body("count", is(2))```.

Reply:
oh nice. I was being lazy

Reply:
it's just an improvement. The biggest thing was broken build.